### PR TITLE
ci: improve setup

### PR DESCRIPTION
**Pull request by manuelnaranjo**
Using a matrix instead of copying the test structure so many times

### CODE REVIEW

**Comment by manuelnaranjo** [comment]
@rmohr this is ready to be reviewed and merged, it reduces how many files I touch on my other PR

File: Makefile [hunk 1]
```diff
@@ -17,9 +17,13 @@ gofmt:
...
 
 e2e:
-	(cd e2e/bazel-5 && bazelisk build //...)
-	(cd e2e/bazel-6 && bazelisk build //...)
-	(cd e2e/bazel-7 && bazelisk build //...)
+	@for version in 5.x 6.x 7.x; do \
+		( \
+			cd e2e/bazel-workspace && \
+			echo "Testing $$version" > /dev/stderr && \
+			USE_BAZEL_VERSION=$$version bazel --batch build //...\
```
Comment (rmohr) [inline Makefile:24]: should this be `bazelisk` instead of `bazel`?

File: .github/workflows/action.yml [hunk 2]
```diff
@@ -6,33 +6,59 @@ on:
...
+        bazelisk-cache: true
+        # Keep a disk-cache
+        disk-cache: true
+        # Share repository cache between workflows.
+        repository-cache: true
+        # enable some flags for CI
+        bazelrc: |
+          import %workspace%/../../.aspect/bazelrc/ci.bazelrc
+          ${{ matrix.version.bazelrc }}
+    - run: cd e2e/bazel-workspace && USE_BAZEL_VERSION=${{ matrix.version.version }} bazel build //...
```
Comment (rmohr) [inline .github/workflows/action.yml:64]: Is this `bazel` binary actually `bazelisk`?

Comment (manuelnaranjo) [reply .github/workflows/action.yml:64]: Yes bazel is a symlink to bazelisk setup by the workflow helpers

File: Makefile (hunk 1 above)
Comment (manuelnaranjo) [reply Makefile:24]: It's already bazelisk, no one uses bazel directly, and in most cases it's linked

Comment (rmohr) [reply Makefile:24]: Yeah I do the same, but still, maybe we can keep it consistent in the file :)

Comment (manuelnaranjo) [reply Makefile:24]: Ohh you mean the makefile uses bazelisk? Didn't notice, will check tomorrow and fix it

File: .github/workflows/action.yml (hunk 2 above)
Comment (manuelnaranjo) [reply .github/workflows/action.yml]: let's make that more explicit across the entire file, I see a point in doing so

File: Makefile (hunk 1 above)
Comment (manuelnaranjo) [reply Makefile]: fixed

**Comment by manuelnaranjo** [comment]
@rmohr alright is updated now and uses bazelisk explicitly both in Makefile and CI

**Pull request merged**